File: 3434-find-the-number-of-distinct-colors-among-the-balls/find-the-number-of-distinct-colors-among-the-balls.cpp
class Solution {
public:
    vector<int> queryResults(int limit, vector<vector<int>>& queries) {
       
        map<int,int>mp;
        vector<int>ans;
        int n=limit+1;
        map<int,int>cnt;
        for(auto it:queries){
            if(mp.find(it[0])==mp.end()){
                mp[it[0]]=it[1];
                
                cnt[it[1]]++;
            }
            else{
             cnt[mp[it[0]]]--;
             if(cnt[mp[it[0]]]==0 )cnt.erase(mp[it[0]]);
             mp[it[0]]=it[1];
             cnt[it[1]]++;
                
            }
            ans.push_back(cnt.size());
           
        }
        return ans;
    }
};

File: 3434-find-the-number-of-distinct-colors-among-the-balls/find-the-number-of-distinct-colors-among-the-balls.py
class Solution:
    def queryResults(self, limit: int, queries: List[List[int]]) -> List[int]:
        mp1={}
        mp2={}
        ans=[]
        n=limit+1
        for x, y in queries:
            if x not in mp1:
                mp1[x]=y
                mp2[y]=mp2.get(y,0)+1
            else:
                col=mp1[x]
                mp2[col]-=1
                if mp2[col]==0:
                    del mp2[col]
                mp1[x]=y
                mp2[y]=mp2.get(y,0)+1
            ans.append(len(mp2))


        return ans
        